app/Policies/GameCommentPolicy.php

```diff
@@ -75,6 +84,15 @@ public function update(User $user, GameComment $comment): bool
 
     public function delete(User $user, GameComment $comment): bool
     {
+        // Handle if comments are locked for this game.
+        if ($comment->commentable && $comment->commentable->comments_locked_at) {
+            // Only moderators and admins can delete when locked.
+            return $user->hasAnyRole([
+                Role::MODERATOR,
+                Role::ADMINISTRATOR,
+            ]);
+        }
```

Comment:
This doesn't seem to be getting called on either the `/game1/` or `/game2/` pages. A user without any of the specified roles is still able to delete their own comments.

I can introduce an exception here and neither page is affected.